custom_components/tuya_local/devices/puppy_kitty_pet_feeder.yaml

```diff
@@ -0,0 +1,67 @@
+name: Pet feeder
+products:
+  - id: iwabnqimdhmzxzvp
+    name: PUPPY KITTY Automatic Cat & Dog Feeder
+primary_entity:
+  entity: switch
+  name: Enable feeder
+  icon: "mdi:food-drumstick"
+  dps:
+    - id: 25
+      type: boolean
+      name: switch
+    - id: 1
+      name: meal_plan
+      type: string
+      optional: true
+secondary_entities:
+  - entity: sensor
+    icon: "mdi:paw"
+    name: Feed report
+    category: diagnostic
+    dps:
+      - id: 14
+        name: sensor
+        type: integer
+        optional: true
+  - entity: button
+    class: restart
+    name: Factory reset
+    category: config
+    dps:
+      - id: 24
+        type: boolean
+        name: button
+        optional: true
+  - entity: number
+    icon: "mdi:paw"
+    name: Manual feed
+    category: config
+    dps:
+      - id: 3
+        name: value
+        type: integer
+        optional: true
+        range:
+          min: 1
+          max: 20
+  - entity: binary_sensor
+    name: Feeding
+    class: running
+    icon: "mdi:paw"
+    category: diagnostic
+    dps:
+      - id: 4
+        type: string
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: feeding
+            value: true
+          - dps_val: standby
+            value: false
+          - dps_val: done
+            value: false
+          - dps_val: null
+            value: null
```

Comment:
I'm not sure this mapping is needed. If the dp is missing (null), a value of false should be OK. Also the mappings for done and standby are redundant, as they map to the default (false).

It may be better to add another attribute using the same dp id, but with the original string value, in case someone finds a need to differentiate the different possibilities for `false`